src/components/okta_sign_in_widget.js

```diff
@@ -0,0 +1,54 @@
+// Okta Sign In Widget Component for React
+
+import React, { Component, PropTypes } from 'react';
+
+export default class OktaSignInWidget extends Component {
+  static contextTypes = {
+    router: PropTypes.object
+  };
+
+  constructor(props) {
+    super(props);
+  }
+
+  componentWillMount() {
+    const that = this;
+
+    // create Okta auth object
+    const auth = new OktaSignIn(this.props.options);
+
+    // check existing session
+    setTimeout(() => {
+      auth.session.exists((exists) => {
+        if(exists) {
+          that.login(that.props.successRoute);
+        } else {
+          that.launchWidget(auth);
+        }
+      });
+    }, 0);
+  }
+
+  login(path) {
+    this.context.router.push(path);
+  }
+
+  launchWidget(auth) {
+    // render the widget and pass it the callback function for login success
+    auth.renderEl({ 
+        el: '#' + this.props.container
```

Comment:
nit: shouldn't hardcode the idea that they will be using an id to specify the container